.clang-tidy

```diff
@@ -62,6 +68,15 @@
 #    code, but we should do that over time, maybe per-directory, instead of
 #    blocking the adoption of Clang 17 until it is all fixed.
 #
+#  -bugprone-unused-local-non-trivial-variable: This might be nice to have, but
```

Comment:
Should we open bugs for these?  FWIW, it is possible to enable (or disable) clang-tidy warnings in some directories, so we could turn these off only on directories where the warnings are common.

Reply:
Yeah that is a good idea. I think I'm gonna split these changes into multiple PRs like you suggested.